src/pages/sources/SourceArticle.tsx

```diff
@@ -0,0 +1,66 @@
+import { Link, useNavigate, useParams } from "react-router-dom";
+import Back from "../../components/Back";
+import { useGetArticleFromSourceQuery } from "../../features/api/apiSlice";
+
+export default function SourceArticle() {
+  const { articleId, sourceId } = useParams();
+  const navigate = useNavigate();
+  const { data: articles } = useGetArticleFromSourceQuery(sourceId);
+  const article = articles?.find((article) => article.id === articleId);
```

Comment:
?

Reply:
The articles can be undefined while fetching.